src/metrics/system/system.go

```diff
@@ -65,6 +67,19 @@ func (s *SystemMetrics) GetTotalCPUStats() gostats.TimesStat {
 	return stats[0]
 }
 
+func (s *SystemMetrics) GetPerCoreUsage(c chan *string) {
+	perCore, err := gostats.Percent(time.Duration(time.Second), true)
+	if err != nil {
+		errors.Unwrap(err)
```

Comment:
I think this is a trivial statement. It returns an error that is never captured. Things like this should be sent via an error channel or use a single channel that accepts an interface and at the receiving side, to the type matching/conversion.